src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/FilingsService.java

```diff
@@ -92,113 +93,117 @@ public FilingsService(
 
     OverseasEntitySubmissionDto submissionDto =
-        setSubmissionData(userSubmission, overseasEntityId, logMap);
-    setPaymentData(userSubmission, transaction, passThroughTokenHeader, logMap);
+            overseasEntitiesService.getOverseasEntitySubmission(overseasEntityId)
+                    .orElseThrow(() -> new SubmissionNotFoundException(
+                            String.format("Empty submission returned when generating filing for %s", overseasEntityId)
+                    ));
 
     if (submissionDto.isForUpdate()) {
-      publicDataRetrievalService.initialisePublicData(
-          submissionDto.getEntityNumber(), passThroughTokenHeader);
-      privateDataRetrievalService.initialisePrivateData(submissionDto.getEntityNumber());
+      initialisePublicAndPrivateData(submissionDto.getEntityNumber(), passThroughTokenHeader);
+      UpdateSubmission updateSubmission = new UpdateSubmission();
+      setUpdateSubmissionData(updateSubmission, submissionDto, logMap);
+      Map<String, Object> updateSubmissionMap = objectMapper.convertValue(updateSubmission, new TypeReference<Map<String, Object>>() {});
+      filing.setData(updateSubmissionMap);
+      filing.setKind(FILING_KIND_OVERSEAS_ENTITY_UPDATE);
+    } else {
+      setSubmissionData(userSubmission, submissionDto, logMap);
+      filing.setData(userSubmission);
+      filing.setKind(FILING_KIND_OVERSEAS_ENTITY);
     }
 
-    filing.setData(userSubmission);
     if (overseasEntitiesService.isSubmissionAnUpdate(requestId, overseasEntityId)) {
       filing.setCost(updateCostAmount);
     } else {
       filing.setCost(registerCostAmount);
     }
 
+    setPaymentData(userSubmission, transaction, passThroughTokenHeader, logMap);
     setDescriptionFields(filing, submissionDto.isForUpdate());
   }
 
-  private OverseasEntitySubmissionDto setSubmissionData(
-      Map<String, Object> data, String overseasEntityId, Map<String, Object> logMap)
-      throws SubmissionNotFoundException, ServiceException {
-    Optional<OverseasEntitySubmissionDto> submissionOpt =
-        overseasEntitiesService.getOverseasEntitySubmission(overseasEntityId);
+  private void initialisePublicAndPrivateData(String entityNumber, String passThroughTokenHeader) throws ServiceException {
+    publicDataRetrievalService.initialisePublicData(entityNumber, passThroughTokenHeader);
+    privateDataRetrievalService.initialisePrivateData(entityNumber);
+  }
+
+  private void setUpdateSubmissionData(UpdateSubmission updateSubmission, OverseasEntitySubmissionDto submissionDto, Map<String, Object> logMap) {
 
-    OverseasEntitySubmissionDto submissionDto =
-        submissionOpt.orElseThrow(
-            () ->
-                new SubmissionNotFoundException(
-                    String.format(
-                        "Empty submission returned when generating filing for %s",
-                        overseasEntityId)));
+    // Populate relevant fields in UpdateSubmission object //
```

Comment:
I think the confusing was caused by setUpdateSubmissionData().  It's now just a one line method that is miss-named if you compare it to setSubmissionData() - which does the inserts into the data map. I'd inline setUpdateSubmissionData() but maybe refactor out a method for the update parts if that makes the setFilingApiData() shorter.

Reply:
Yeah that line of code was more of a placeholder, I've removed it to avoid confusion. This method is currently just a shell, its for the other tickets to fill and update their models from, so setFilingApiData() doesn't get cluttered.

Reply:
I'd still inline it and let the follow on PR refactor. Probably a bigger piece of setFilingApiData() could be broken out - with a more sensible name in next PR.